tests/test_converters/test_obo_graph_to_fhir.py

```diff
@@ -10,28 +13,74 @@
 from tests import IMBO, INPUT_DIR, NUCLEUS, OUTPUT_DIR
 from tests.test_implementations import ComplianceTester
 
-ONT = INPUT_DIR / "go-nucleus.json"
-OUT = OUTPUT_DIR / "go-nucleus.fhir.codesystem.json"
-
 
 class OboGraphToFHIRTest(unittest.TestCase):
-    """Tests OBO JSON -> FHIR."""
+    """Tests OBO JSON -> FHIR. Different ontologies have unique structures, so test some specifics for those."""
+
+    def _load_ontology(self, url: str, download_path: str, use_cache: bool = True) -> GraphDocument:
+        """Downloads ontology if needed, and loads it."""
+        if not os.path.exists(download_path) and use_cache:
+            r = requests.get(url)
+            with open(download_path, 'wb') as f:
+                f.write(r.content)
+            return json_loader.load(str(download_path), target_class=GraphDocument)
+        elif use_cache:
+            return json_loader.load(str(download_path), target_class=GraphDocument)
+        return json_loader.load(url, target_class=GraphDocument)
 
     def setUp(self):
+        """Set up tests"""
         self.converter = OboGraphToFHIRConverter()
         self.converter.curie_converter = curies.Converter.from_prefix_map(get_default_prefix_map())
         self.compliance_tester = ComplianceTester(self)
 
-    def test_convert(self):
-        """Tests parsing then converting to fhir documents."""
-        gd: GraphDocument = json_loader.load(str(ONT), target_class=GraphDocument)
-        self.converter.dump(gd, OUT, include_all_predicates=True)
-        cs: CodeSystem
-        cs = json_loader.load(str(OUT), target_class=CodeSystem)
+    def test_convert_go_nucleus(self):
+        """General test & specifis for go nucleus."""
+        ont = INPUT_DIR / "go-nucleus.json"
+        out = OUTPUT_DIR / "CodeSystem-go-nucleus.json"
+        gd: GraphDocument = json_loader.load(str(ont), target_class=GraphDocument)
+        self.converter.dump(gd, out, include_all_predicates=True)
+        cs: CodeSystem = json_loader.load(str(out), target_class=CodeSystem)
         self.assertEqual("CodeSystem", cs.resourceType)
         [nucleus_concept] = [c for c in cs.concept if c.code == NUCLEUS]
         self.assertEqual("nucleus", nucleus_concept.display)
         self.assertTrue(nucleus_concept.definition.startswith("A membrane-bounded organelle"))
         parents = [x for x in nucleus_concept.property if x.code == "parent"]
         self.assertEqual(len(parents), 1)
         self.assertTrue(parents[0].valueCode == IMBO)
+
+    def test_convert_mondo(self):
```

Comment:
I think I'll need to test cases specific to each ontology, so felt like a method for each one was best.